src/plugins/calibration/include/plugins/calibration/calibration.h

```diff
@@ -35,99 +43,122 @@ class Calibration : public PluginBase {
     ~Calibration();
 
     /**
-     * @brief Possible results returned for calibration commands.
+     * @brief Possible results returned for calibration commands
      */
     enum class Result {
-        UNKNOWN,
-        SUCCESS,
-        IN_PROGRESS,
-        INSTRUCTION,
-        FAILED,
-        NO_SYSTEM,
-        CONNECTION_ERROR,
-        BUSY,
-        COMMAND_DENIED,
-        TIMEOUT,
-        CANCELLED
+        Unknown, /**< @brief Unknown error. */
+        Success, /**< @brief The calibration process succeeded. */
+        InProgress, /**< @brief Intermediate message showing progress of the calibration process. */
+        Instruction, /**< @brief Intermediate message giving instructions on the next steps required
+                        by the process. */
+        Failed, /**< @brief Calibration failed. */
+        NoSystem, /**< @brief No system is connected. */
+        ConnectionError, /**< @brief Connection error. */
+        Busy, /**< @brief Vehicle is busy. */
+        CommandDenied, /**< @brief Command refused by vehicle. */
+        Timeout, /**< @brief Command timed out. */
+        Cancelled, /**< @brief Calibration process got cancelled. */
     };
 
+    /**
+     * @brief Stream operator to print information about a `Calibration::Result`.
+     *
+     * @return A reference to the stream.
+     */
+    friend std::ostream& operator<<(std::ostream& str, Calibration::Result const& result);
+
     /**
      * @brief Progress data coming from calibration.
      *
      * Can be a progress percentage, or an instruction text.
      */
     struct ProgressData {
-        bool has_progress; ///< Whether or not this struct contains progress information.
-        float progress; ///< The actual progress.
-        bool has_status_text; ///< Whether or not this struct contains an instruction text.
-        std::string status_text; ///< The actual instruction text.
-
-        /**
-         * Constructor for ProgressData.
-         *
-         * @param _has_progress Whether or not this struct contains progress information.
-         * @param _progress The actual progress.
-         * @param _has_status_text Whether or not this struct contains an instruction text.
-         * @param _status_text The actual instruction text.
-         */
-        ProgressData(
-            const bool _has_progress,
-            const float _progress,
-            const bool _has_status_text,
-            const std::string& _status_text) :
-            has_progress(_has_progress),
-            progress(_progress),
-            has_status_text(_has_status_text),
-            status_text(_status_text)
-        {}
+        bool has_progress{
+            false}; /**< @brief Whether this ProgressData contains a 'progress' status or not */
```

Comment:
With C++17 we could make this `std::optional<float> progress`.

Reply:
Sure